### ISSUE TITLE
regression with recent jlab

### ISSUE THREAD

**Issue by parmentelat** [parmentelat/jupyterlab-gridwidth#16]
for a couple days/weeks now, I have noticed a regression in jlab sessions that use this extension

the symptoms are

- cells take needlessly large height
- scrolling up and down triggers odd behaviours

and this even with notebooks that do not have gridwidth tags

I need to
- [ ] investigate and confirm my gut feeling that a recent jlab is causing the issue
- [ ] bisect to find the actual version that breaks
- [ ] find a way to fix the problem

because at first sight what is happening it it rather mysterious

<details><summary>jupyter --version</summary>

```bash
Selected Jupyter core packages...
IPython          : 8.25.0
ipykernel        : 6.29.4
ipywidgets       : 8.1.3
jupyter_client   : 8.6.2
jupyter_core     : 5.7.2
jupyter_server   : 2.14.1
jupyterlab       : 4.2.1
nbclient         : 0.10.0
nbconvert        : 7.16.4
nbformat         : 5.10.4
notebook         : 7.2.0
qtconsole        : 5.5.2
traitlets        : 5.14.3
```

</details>

**Comment by parmentelat** [comment]
when forcing `pip install jupyterlab<4.1` I get the expected behaviour, i.e. this, on the first example notebooks in this repo

![gridwidth-ok](https://github.com/parmentelat/jupyterlab-gridwidth/assets/4519227/c16b6331-64b0-44a4-ab4f-d79dc35df6a2)

**Comment by parmentelat** [comment]
here's the behaviour obtained on the same notebook with jupyterlab 4.2.1
the whole time we're only scrolling down with the mouse

![gridwidth-ko](https://github.com/parmentelat/jupyterlab-gridwidth/assets/4519227/7239bc43-a95a-4f90-955e-215f027e0240)

**Comment by krassowski** [comment]
Can you go to Notebook settings and change the windowing mode to `none` from `full` (default since 4.2) and see what happens? Similarly, on 4.1 if you change to `full` can you reproduce this? I would suggest reloading JupyterLab after changing this setting.

**Comment by parmentelat** [comment]
with 4.2.1, if I go to 'none' windowing mode, the issue vanishes

what's the deal with the windowing mode ?

**Comment by parmentelat** [comment]
and conversely under 4.1.8, moving to 'full` windowing mode triggers the issue

**Comment by parmentelat** [comment]
and conversely under 4.1.8, moving to 'full` windowing mode triggers the issue

**Comment by parmentelat** [comment]
is there a way for an extension to act on these settings ?

**Comment by krassowski** [comment]
Yes, jupyterlab-execute-time does so, but it has a setting specific to its use case. Some users may find it annoying if you switch the windowing off but as a short-term solution until you can figure out how to support windowed mode it sounds reasonable to me (provided it is well documented).

**Comment by parmentelat** [comment]
thanks @krassowski; about this:
> until you can figure out how to support windowed mode
I wouldn't know where to start, where is this 'windowing' thing documented ?

**Comment by krassowski** [comment]
> I wouldn't know where to start, where is this 'windowing' thing documented ?

https://jupyterlab.readthedocs.io/en/stable/developer/performance.html#windowed-list

https://jupyterlab.readthedocs.io/en/stable/api/modules/ui_components.WindowedList.html

https://jupyterlab.readthedocs.io/en/stable/api/classes/notebook.NotebookWindowedLayout.html

https://jupyterlab.readthedocs.io/en/stable/api/classes/notebook.NotebookViewModel.html

**Comment by parmentelat** [comment]
thanks for the pointers

in the meanwhile I'm tempted by the short-term approach (because otherwise the disease will slowly spread to all my students as they upgrade their envs, leading to a mess..)

it feels however that it's not just a matter of changing the `windowingMode` setting directly - because understandbly my extension won't be allowed to modify the notebook's extension settings, iiuc from reading this
https://discourse.jupyter.org/t/how-do-i-override-settings-using-my-extensions-schemadir-file/8599/2

now, how can I achieve this as an externion developer ?
as opposed to an operator, which the above post only addresses - iiuc again

thanks !

**Comment by krassowski** [comment]
Does this help?

https://github.com/deshaw/jupyterlab-execute-time/blob/c020582313868db92f1b0bdbae71aeb305b29fc1/src/index.ts#L54-L66

**Comment by parmentelat** [comment]
yes thanks, I believe this should get me started

just a question though
do I understand correctly that in the linked code, 'enabled' is a setting local to the execute-time extension, and it is not related to the generic 'enabled/disabled' status of an extension, is that right ?
I would have thought that the `activate` method of the plugin would not even be triggered if the extension is disabled, no ? (in which case a test against 'enabled' would always return true...)
